core/src/test/scala/com/microsoft/azure/synapse/ml/codegen/RTestGen.scala

```diff
@@ -0,0 +1,151 @@
+      s"""|Package: ${conf.name.replace("-", ".")}
+          |Title: Access to SynapseML via R
+          |Description: Provides an interface to SynapseML.
+          |Version: ${conf.rVersion}
+          |Date: $today
+          |Author: Microsoft Corporation
+          |Maintainer: SynapseML Team <synapseml-support@microsoft.com>
+          |URL: https://github.com/Microsoft/SynapseML
+          |BugReports: https://github.com/Microsoft/SynapseML/issues
+          |Depends:
+          |    R (>= 3.5.0)
+          |Imports:
+          |    sparklyr
+          |License: MIT
+          |Suggests:
+          |    testthat (>= 3.0.0)
+          |Config/testthat/edition: 3
+          |""".stripMargin)
+
+    val scalaVersion = BuildInfo.scalaVersion.split(".".toCharArray).dropRight(1).mkString(".")
+    writeFile(new File(projectDir, "synapsemltest.Rproj"),
+      """
+        |Version: 1.0
+        |
+        |RestoreWorkspace: Default
+        |SaveWorkspace: Default
+        |AlwaysSaveHistory: Default
+        |
+        |EnableCodeIndexing: Yes
+        |UseSpacesForTab: Yes
+        |NumSpacesForTab: 4
+        |Encoding: UTF-8
+        |
+        |RnwWeave: Sweave
+        |LaTeX: pdfLaTeX
+        |
+        |BuildType: Package
+        |PackageUseDevtools: Yes
+        |PackageInstallArgs: --no-multiarch --with-keep.source
+        |
+        |""".stripMargin)
+
+    val synapseVersion = BuildInfo.version
+
+    writeFile(join(conf.rTestThatDir, "setup.R"),
+      s"""
+         |if (!require("sparklyr")) {
+         |  install.packages("sparklyr")
+         |  library("sparklyr")
+         |}
+         |
+         |options(sparklyr.log.console = TRUE)
+         |options(sparklyr.verbose = TRUE)
+         |
+         |conf <- spark_config()
+         |conf$$sparklyr.shell.conf <- c(
+         |  "spark.app.name=RSparkTests",
+         |  "spark.jars.packages=com.microsoft.azure:synapseml_2.12:${conf.version}",
+         |  "spark.jars.repositories=https://mmlspark.azureedge.net/maven",
+         |  "spark.executor.heartbeatInterval=60s",
+         |  "spark.sql.shuffle.partitions=10",
+         |  "spark.sql.crossJoin.enabled=true")
+         |
+         |sc <- spark_connect(
+         |  master = "local",
+         |  version = "3.2.0",
+         |  config = conf)
+         |
+         |""".stripMargin)
+
+    val library = conf.name.replaceAll("-", ".")
+    writeFile(join(testsDir, "testthat.R"),
+      s"""
+         |${useLibrary("testthat")}
+         |${useLibrary("jsonlite")}
+         |${useLibrary("mlflow")}
+         |library($library)
+         |
+         |""".stripMargin)
+
+  }
+
+  def useLibrary(name: String): String = {
+    s"""
+       |if (!require("${name}")) {
+       |  install.packages("${name}")
+       |  library("${name}")
+       |}
+       |""".stripMargin
+  }
+
+  def isDeprecated(name: String): Boolean = {
+    name.contains("LIME") && !name.toLowerCase.contains("explainer")
+  }
```

Comment:
Hopefully removing the R test gen trait will allow you to remove this logic

Reply:
See my reply in CodeGen.scala